ckb_identity.h

```diff
@@ -0,0 +1,353 @@
+#ifndef CKB_C_STDLIB_CKB_IDENTITY_H_
+#define CKB_C_STDLIB_CKB_IDENTITY_H_
+
+#include <blake2b.h>
+#include <ckb_exec.h>
+
+#include "ckb_consts.h"
+
+#define CKB_IDENTITY_LEN 21
+#define RECID_INDEX 64
+#define ONE_BATCH_SIZE 32768
+#define PUBKEY_SIZE 33
+#define MAX_WITNESS_SIZE 32768
+#define BLAKE2B_BLOCK_SIZE 32
+#define BLAKE160_SIZE 20
+#define SECP256K1_SIGNATURE_SIZE 65
+#define SECP256K1_MESSAGE_SIZE 32
+#define MAX_PREIMAGE_SIZE 1024
+#define MAX_CODE_SIZE (1024 * 400)
+
+enum CkbIdentityErrorCode {
+  ERROR_IDENTITY_ARGUMENTS_LEN = -1,
+  ERROR_IDENTITY_ENCODING = -2,
+  ERROR_IDENTITY_SYSCALL = -3,
+
+  // compatible with secp256k1 pubkey hash verification
+  ERROR_IDENTITY_SECP_RECOVER_PUBKEY = -11,
+  ERROR_IDENTITY_SECP_PARSE_SIGNATURE = -14,
+  ERROR_IDENTITY_SECP_SERIALIZE_PUBKEY = -15,
+  ERROR_IDENTITY_PUBKEY_BLAKE160_HASH = -31,
+  // new error code
+  ERROR_IDENTITY_LOCK_SCRIPT_HASH_NOT_FOUND = 70,
+  ERROR_IDENTITY_WRONG_ARGS,
+  ERROR_INVALID_PREIMAGE,
+};
+
+typedef struct CkbIdentityType {
+  uint8_t flags;
+  // blake160 (20 bytes) hash of lock script or pubkey
+  uint8_t blake160[20];
+} CkbIdentityType;
+
+enum IdentityFlagsType {
+  IdentityFlagsPubkeyHash = 0,
```

Comment:
Let's rename this to `IdentityFlagsCkb`, since it's actually CKB's way of verifying a signature. All the values from 0 through 5 actually would all assume identity contains pubkey hash